### PR TITLE
Enable save changes button only after labels added for all app

### PR DESCRIPTION
**Pull request by TimothyAsirJeyasing**

### CODE REVIEW

**Comment by TimothyAsirJeyasing** [comment]
https://user-images.githubusercontent.com/6670284/219623773-a4a3b88a-1cd5-4b1e-9ac2-933caa5acd16.mp4

**Comment by TimothyAsirJeyasing** [comment]
@SanjalKatiyar Please review

File: packages/mco/components/modals/drpolicy-apps-apply/apply-policy-modal.tsx [hunk 1]
```diff
@@ -88,6 +88,19 @@ const ApplyDRPolicyModal: React.FC<CommonModalProps<ApplyModalExtraProps>> = (
...
+    if (
+      protectedResources.some((protectedResource: PlacementToAppSets) => {
+        const updateLabels =
+          drpcPvcLabels?.[protectedResource.namespace]?.[
+            protectedResource.placement
+          ]?.updateLabels || [];
+        return updateLabels.length === 0;
+      })
+    ) {
+      return true;
```
Comment (GowthamShanmugam) [inline packages/mco/components/modals/drpolicy-apps-apply/apply-policy-modal.tsx:101]: Returning true from here will skip all other mandatory checks

**Comment by GowthamShanmugam** [comment]
> 

Do not allow label change for already protected applications, i think i have added this check, please rebase and try this once again.

File: packages/mco/components/modals/drpolicy-apps-apply/apply-policy-modal.tsx [hunk 2]
```diff
@@ -88,6 +88,19 @@ const ApplyDRPolicyModal: React.FC<CommonModalProps<ApplyModalExtraProps>> = (
     const drpcPvcLabels: PlacementToDrpcMap =
       state.drpcPvcLabels[state.appType];
 
+    // "Save changes" to be enabled only if all the resources on protected panel have any label assigned
+    if (
+      protectedResources.some((protectedResource: PlacementToAppSets) => {
+        const updateLabels =
```
Comment (GowthamShanmugam) [inline packages/mco/components/modals/drpolicy-apps-apply/apply-policy-modal.tsx:94]: Does this check only for !protectedResource?.isAlreadyProtected (we should not allow label edit for already protected apps, so we need to do this check for the apps which is newly selected for apply policy )

Comment (GowthamShanmugam) [reply packages/mco/components/modals/drpolicy-apps-apply/apply-policy-modal.tsx:94]: it is temporary check due to DR backend issue

Comment (TimothyAsirJeyasing) [reply packages/mco/components/modals/drpolicy-apps-apply/apply-policy-modal.tsx:101]: updated to enable only when new app moved to protected resource with label.
feeling like disable expand hide the info about what are the label added to the app already.

File: packages/mco/components/modals/drpolicy-apps-apply/apply-policy-modal.tsx [hunk 3]
```diff
@@ -106,25 +136,8 @@ const ApplyDRPolicyModal: React.FC<CommonModalProps<ApplyModalExtraProps>> = (
         },
       });
     if (disableSubmit === false) return disableSubmit;
-
-    // if any non-protected resource present on protected panel => we need to create DRPC, enable "Save changes" in this case
-    // if any new PVC label added to any resource => we need to update DRPC, enable "Save changes" in this case
-    disableSubmit = !protectedResources.some(
```
Comment (GowthamShanmugam) [inline packages/mco/components/modals/drpolicy-apps-apply/apply-policy-modal.tsx:112]: I can see a lot of (if) conditions in the above logic, we can still use the below logic by tuning ```areLabelsDifferent``` different function.

we can reuse functions ```areLabelsPresent``` and with one extra condition is checking label count is not empty.

Also, remove or  comment out the following section from submit function:
// 2. updating PVC labels only if: DRPC is neither deleted nor created in previous steps

Comment (GowthamShanmugam) [reply packages/mco/components/modals/drpolicy-apps-apply/apply-policy-modal.tsx:112]: We dont need extra code this 109 to 127 code will work with minor tuning in areLabelsPresent function

**Comment by TimothyAsirJeyasing** [comment]
https://user-images.githubusercontent.com/6670284/221554365-b45893ef-9418-43e2-958f-a2fc45e7b7c5.mp4

**Comment by SanjalKatiyar** [comment]
talk to UX first on how to show users that labels are required field... just blocking the button without telling user any reason for that is not correct...

**Comment by TimothyAsirJeyasing** [comment]
https://user-images.githubusercontent.com/6670284/222404026-cfc72360-8d2d-4249-9282-17340b171f9f.mp4

File: packages/mco/components/modals/drpolicy-apps-apply/apply-policy-dual-list-selector.scss [hunk 4]
```diff
@@ -7,4 +7,7 @@
     &__ItemDescription--color {
         color: var(--pf-global--Color--200);
     }
+    &__pvcselector {
+        margin-top: var(--pf-global--spacer--md);
```
Comment (GowthamShanmugam) [inline packages/mco/components/modals/drpolicy-apps-apply/apply-policy-dual-list-selector.scss:11]: Why this change?

File: packages/mco/components/modals/drpolicy-apps-apply/apply-policy-modal.tsx [hunk 5]
```diff
@@ -90,12 +84,40 @@ const ApplyDRPolicyModal: React.FC<CommonModalProps<ApplyModalExtraProps>> = (
...
+        [[], true]
+      );
+
+    const changesFoundOnRightSide = !!applicationsToEnable.length;
+    const noIssuesFoundOnRightSide = isLablesAreNotEmpty;
+    const changesFoundOnLeftSide = !!applicationsToDisable;
+    // we dont have any left side check, By default keeping it as true
+    const noIssuesFoundOnLeftSide = true;
+
+    if (state.message.text && !changesFoundOnLeftSide) state.message.text = '';
```
Comment (GowthamShanmugam) [inline packages/mco/components/modals/drpolicy-apps-apply/apply-policy-modal.tsx:118]:
```suggestion
   
```

Comment (GowthamShanmugam) [reply packages/mco/components/modals/drpolicy-apps-apply/apply-policy-modal.tsx:118]: I know you are doing this to disable the warning message. But it looks like we need to tink more. For now, remove it. let us think the proper way to clean the message.

File: packages/mco/components/modals/drpolicy-apps-apply/apply-policy-dual-list-selector.scss (hunk 4 above)
Comment (GowthamShanmugam) [reply packages/mco/components/modals/drpolicy-apps-apply/apply-policy-dual-list-selector.scss:11]: can you elaborate on this more? where exactly it provides equal space. is space between the expanded view and label selector is more?

Comment (TimothyAsirJeyasing) [reply packages/mco/components/modals/drpolicy-apps-apply/apply-policy-dual-list-selector.scss:11]: I expect it will give some standard 16px space between "Add label" and the textbox. But here it is showing the default space only. Not been controlled by this change. So i feel this is not required.

File: packages/mco/components/modals/drpolicy-apps-apply/apply-policy-modal.tsx [hunk 6]
```diff
@@ -199,27 +203,27 @@ const ApplyDRPolicyModal: React.FC<CommonModalProps<ApplyModalExtraProps>> = (
...
-            drpcPvcLabels?.[protectedResource.namespace]?.[
-              protectedResource.placement
-            ]?.updateLabels || [];
-          // update only if original and updated labels are different
-          areLabelsDifferent(existingLabels, updateLabels) &&
-            promises.push(
-              ...getUpdatedDRPCPromise(protectedResource, drpcPvcLabels)
-            );
-        }
+        // // 2. updating PVC labels only if: DRPC is neither deleted nor created in previous steps
```
Comment (GowthamShanmugam) [inline packages/mco/components/modals/drpolicy-apps-apply/apply-policy-modal.tsx:206]: use multi-line comment /*comment here*/ with TODO message:
// TODO: Allow PVC selector update when the ramen is supporting

Comment (TimothyAsirJeyasing) [reply packages/mco/components/modals/drpolicy-apps-apply/apply-policy-modal.tsx]: Done

**Comment by TimothyAsirJeyasing** [comment]
https://user-images.githubusercontent.com/6670284/222546094-4e5a9e51-405a-4b78-b329-0d75eb2d8751.mp4

**Review by GowthamShanmugam — APPROVED** [review]

**Review by GowthamShanmugam — APPROVED** [review]

**Comment by GowthamShanmugam** [comment]
@TimothyAsirJeyasing This Pr needs BZ
